src/socket.rs

```diff
@@ -1463,14 +1464,14 @@ impl Connection {
                     let max_sample = cmp::max(prev_average_delay, self.average_delay);
 
                     if min_sample > 0 {
-                        self.average_delay_base += min_sample as u32;
+                        self.average_delay_base =
+                            self.average_delay_base.wrapping_add(min_sample as u32);
```

Comment:
I don't know a lot about this algorithm, but does it really make sense to use wrapping arithmetic here? You're saying that if the delay gets big enough it suddenly becomes zero. That doesn't seem right.